### PR TITLE
multiple mocks

### PR DESCRIPTION
**Pull request by xieyunzi**
https://github.com/jjh42/mock/issues/44

### CODE REVIEW

**Comment by xieyunzi** [comment]
hello @jjh42

File: lib/mock.ex [hunk 1]
```diff
@@ -55,6 +55,41 @@ defmodule Mock do
     end
   end
 
+
+  @doc """
+  Mock up multiple module for the duration of `test`.
```
Comment (jjh42) [inline lib/mock.ex]: minor spelling - module -> modules.

File: lib/mock.ex [hunk 2]
```diff
@@ -55,6 +55,41 @@ defmodule Mock do
...
+
+  ## Example
+  with_mocks([{HTTPotion, opts, [{get: fn("http://example.com") -> "<html></html>" end}]}]) do
+    # Tests that make the expected call
+    assert called HTTPotion.get("http://example.com")
+  end
+  """
+  defmacro with_mocks(mocks, do: test) do
```
Comment (jjh42) [inline lib/mock.ex]: a lot of this is redundant with with_mock. Probably `with_mock` should now just call `with_mocks`

**Comment by jjh42** [comment]
Hi. Sorry I missed this. I've added a couple of comments. Also, could you add something to the README.

Thanks,

**Comment by xieyunzi** [comment]
Hi @jjh42 , it's been updated.

**Comment by jjh42** [comment]
Thanks @xieyunzi for updating the README. The other suggestion I had was to refactor `with_mock` to simply call `with_mocks`. Right now there is a lot of redundancy between those two macros.

**Comment by xieyunzi** [comment]
You mean remove `with_mock` ?

**Comment by jjh42** [comment]
not remove since we need backwards compatibility, refactor be implemented
by calling with_mocks.

J

**Comment by xieyunzi** [comment]
Haha 😅 miss understand, updated now

**Comment by jjh42** [comment]
Thanks. Great contribution. I'll try and do a new release soon.

**Comment by safwank** [comment]
Any update on the new release?

**Comment by JGulbronson** [comment]
+1 Would also like to be able to use with_mocks

**Comment by safwank** [comment]
Bueller?

**Comment by matteosister** [comment]
@jjh42 could you please tag a new release?

**Comment by jjh42** [comment]
Hi. Sorry I've been swamped, will try and get to this shortly.

**Comment by jjh42** [comment]
This has been published to hex now. Hopefully that resolves your problems. Please let me know how you get on.

**Pull request merged**